python-api/livy/client.py

```diff
@@ -0,0 +1,167 @@
+        :param job: A function that has JobContext as the parameter. Spark jobs can
+            be created with the help of JobContext, which exposes the Spark libraries
+        :type job: A named or anonymous function that has type JobContext as the parameter
+        :returns A future to monitor the status of the job
+        """
+        pass
+
+    def add_file(self, file_uri):
+        """
+        Adds a file to the running remote context.
+
+        Note that the URL should be reachable by the Spark driver process. If running the driver
+        in cluster mode, it may reside on a different host, meaning "file:" URLs have to exist
+        on that node (and not on the client machine).
+
+        :param file_uri: Representation of the path to a file using the filesystem encoding
+        :type file_uri: PurePath from pathlib or pathlib2(preferred)
+
+        >>> import pathlib2
+        >>> pathlib2.Path("/test.txt").asuri()
+
+        :returns A future to monitor the status of the job
+        """
+        pass
+
+    def add_pyfile(self, file_uri):
+        """
+        Adds a .py or .zip to the running remote context.
+
+        Note that the URL should be reachable by the Spark driver process. If running the driver
+        in cluster mode, it may reside on a different host, meaning "file:" URLs have to exist
+        on that node (and not on the client machine).
+
+        :param file_uri: Representation of the path to a file using the filesystem encoding
+        :type file_uri: PurePath from pathlib or pathlib2(preferred)
+
+        >>> import pathlib2
+        >>> pathlib2.Path("/test.txt").asuri()
+
+        :returns A future to monitor the status of the job
+        """
+        pass
+
+    def upload_file(self, open_file):
+        """
+        Upload a file to be passed to the Spark application
+        :param file open_file: The local file to be uploaded
+        :returns A future that can be used to monitor this operation
+        """
+        pass
+
+    def upload_pyfile(self, open_file):
+        """
+        Upload a .py or .zip dependency to be passed to the Spark application
+        :param file open_file: The local file to be uploaded
+        :returns A future that can be used to monitor this operation
+        """
+        pass
+
+    def __set_uri(self, uri):
+        if uri is not None and ('http' == uri.scheme or 'https' == uri.scheme):
+            self.__config.set(self.__CONFIG_SECTION, 'livy.uri', uri.geturl())
+        else:
+            url_exception = uri.geturl if uri is not None else None
+            raise Exception('Cannot create client - Uri not supported - ', url_exception)
+
+    def __set_conf(self, key, value):
+        if value is not None:
+            self.__config.set(self.__CONFIG_SECTION, key, value)
+        else:
+            self.__config.remove_option('env', key)
+
+    def __set_multiple_conf(self, conf_dict):
+        if conf_dict is not None:
```

Comment:
It would be better to not call this function at all if you don't have a config dict